src/makeInfinityAdder.js

```diff
@@ -6,7 +6,23 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
-}
+  return function add(...args) {
+    if (args.length === 0) {
+      return 0;
+    }
+
+    let result = args[0];
+
+    return function adder(...newArgs) {
+      if (newArgs.length !== 0) {
```

Comment:
pay attention that since we work with only one number in params, you can use check for  empty value, not the length